### DISCUSSION TITLE
Font at xref not present

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#3523] [Looking for help] [answered]
Tried changing the color of the first line of text using page.insert_text and by applying redactions. Got an error "font at xref 5 not present". Could you let me know how to resolve this.
[Uploading FireEmergencyGuide_untagged 1.pdf…]()

**Comment by meghanaviyyapu** [comment]
[FireEmergencyGuide_untagged 1.pdf](https://github.com/pymupdf/PyMuPDF/files/15450398/FireEmergencyGuide_untagged.1.pdf)

**Reply by JorjMcKie** [reply]
This font is not embedded - did you look at the output of `page.get_fonts()` and see the "n/a"?
Therefore it cannot be extracted of course.
So you must choose some other font instead.

**Reply by meghanaviyyapu** [reply]
Thanks a lot. Yeah looked at the n/a. Is there a way to insert using the same font irrespective of n/a? What could be the reason for the output being n/a?

**Reply by JorjMcKie** [reply, marked as answer]
Only the PDF creator knows why he did this!
You have to guess and try which file-based font might be right. Or maybe just use "Times-Roman" (fontname="tiro") and hope for the best.

**Reply by JorjMcKie** [reply]
or "tibo" - just saw the "bold"

**Reply by meghanaviyyapu** [reply]
Ok thanks a lot for the explanation
